### ISSUE TITLE
Binary Coded Ascent Algorithm implementation

### ISSUE THREAD

**Issue by vijaysrinivastida** [scikit-learn-contrib/scikit-learn-contrib#26]
Request for project inclusion in scikit-learn-contrib

* Project name: Binary Coordinate Ascent algorithm
* Project description: It is an efficient optimization technique for feature subset selection in machine learning. With the help of this algorithm the efficiency in number of subset selection of available increased by a factor of 5-37 compared with two popular algorithms sequential forward selection (SFS) and sequential floating forward selection (SFFS) while classification of unseen data is maintained without loss of performace.
Reference: Amin Zarshenas and Suzuki K.:  Binary Coordinate Ascent: An efficient optimization technique for feature subset selection for machine learning, Knowledge-Based Systems: 110: 191-201, 2016.
* Authors: Amin Zarshenas , Kenji Suzuki
* Current repository: https://github.com/vijaysrinivastida/bcaalgorithm.git
* Requirements:
 * [ ] scikit-learn compatible (check_estimator passed)
 * [ ] Documentation (guide, API reference, example gallery)
 * [ ] Unit tests (coverage: xx%)
 * [ ] Python3 compatible
 * [ ] PEP8 compliant
 * [ ] Continuous integration

**Comment by fabianp** [comment]
current repository is empty?

**Comment by vijaysrinivastida** [comment]
Yeah I’ll work on it if the topic is accepted ?

**Comment by glemaitre** [comment]
We usually accept project which are working and implemented.

**Comment by jnothman** [comment]
Please reopen when you have a working and tested and documented implementation

**Comment by vijaysrinivastida** [comment]
Hi Joel,
My name is Vijay. Me and my instructor implementing binary coded ascent algorithm for feature selection. Is it necessary for the implemented code to pass check_estimator step to contribute code? Also any additional steps required to submit the final implemented code ?
Thanks and regards 
Vijay

**Comment by glemaitre** [comment]
Hi Vinjay. You actually need to pass the estimator_checks tests. It ensure that people can use your feature selector within pipeline and any sklearn component.

**Comment by vijaysrinivastida** [comment]
Estimator_check tests where I can find this? I saw check_estimator is for classifiers right ? 
Thanks and regards 
Vijay

**Comment by glemaitre** [comment]
Estimators are the base class of transformer, classifier, regressor, etc. So you need to be sure that your custom estimator pass this test: http://scikit-learn.org/stable/modules/generated/sklearn.utils.estimator_checks.check_estimator.html

**Comment by vijaysrinivastida** [comment]
My implementation is with algorithm and it uses the classifiers with in sklearn library even though my algorithm definitely should pass this test. 
Thanks and regards

**Comment by glemaitre** [comment]
So you should write the test ensuring that this is working. This is part of the requirements.

**Comment by vijaysrinivastida** [comment]
Hi I also had question regarding  the process for maintaining the
implemented code and solving the future bugs and issues?
Thanks
Vijay

**Comment by jnothman** [comment]
You would retain the primary maintainer.

**Comment by vijaysrinivastida** [comment]
Hi,
I have doubt regarding continuous integration. My code is implementation of
algorithm and I wrote my own test case it is working.  Is continuous
integration required for my code implementation as I didn't use classifier?
Thanks and regards
Vijay

**Comment by jnothman** [comment]
It's very hard for us to check something if you don't show the code.

I would think that if your implementation is as standard an interface as a feature selector then, yes, it should pass check_estimator completely.
